detector.go

```diff
@@ -195,32 +209,55 @@ func (d *Detector) detectGroup(group buildpack.Group, done []buildpack.GroupElem
 			return d.detectOrder(groupEl.OrderExtensions, done, group.Group[i+1:], true, wg)
 		}
 
-		// Lookup element in store.
+		// Lookup element in store.  <-- "the store" is the directory where all the buildpacks are.
 		var (
-			descriptor buildpack.Descriptor
-			err        error
+			descriptor   buildpack.Descriptor
+			bpDescriptor *buildpack.BpDescriptor
+			err          error
 		)
 		if groupEl.Kind() == buildpack.KindBuildpack {
-			descriptor, err = d.DirStore.LookupBp(groupEl.ID, groupEl.Version)
+			bpDescriptor, err = d.DirStore.LookupBp(groupEl.ID, groupEl.Version)
 			if err != nil {
 				return nil, nil, err
 			}
 
+			targetMatch := false
+			if d.AnalyzeMD.RunImage.Target.IsWildcard() {
+				targetMatch = true
+			} else {
+				for _, target := range bpDescriptor.Targets {
```

Comment:
Is the idea that buildpacks won't have targets unless they are opting into this validation? Does this validation need to be guarded at a platform API level?

Reply:
hmm good point.  I was writing thinking that they'd all be empty/wildcard, but i guess there's a case where one of them would get the bionic stack and the other wouldn't, so then it wouldn't satisfy even if somebody was trying to run with an older API which would be a new bug/regression from user perspective,  so anyways you're right. API guard is the right call.  thanks!